test/stablecoin_converter.js

```diff
@@ -1249,24 +1249,23 @@ contract("StablecoinConverter", async (accounts) => {
   })
 })
 
-const HEX_WORD_SIZE = 64
 function decodeAuctionElements(bytes) {
-  bytes = bytes.slice(2)
   const result = []
+  bytes = bytes.slice(2) // cutting of 0x
   while (bytes.length > 0) {
-    const element = bytes.slice(0, HEX_WORD_SIZE * 10)
-    bytes = bytes.slice(HEX_WORD_SIZE * 10)
+    const element = bytes.slice(0, 113 * 2)
+    bytes = bytes.slice(113 * 2)
     result.push({
-      user: "0x" + element.slice(HEX_WORD_SIZE - 40, HEX_WORD_SIZE), // address is only 20 bytes
-      sellTokenBalance: parseInt(element.slice(1 * HEX_WORD_SIZE, 2 * HEX_WORD_SIZE), 16),
-      buyToken: parseInt(element.slice(2 * HEX_WORD_SIZE, 3 * HEX_WORD_SIZE), 16),
-      sellToken: parseInt(element.slice(3 * HEX_WORD_SIZE, 4 * HEX_WORD_SIZE), 16),
-      validFrom: parseInt(element.slice(4 * HEX_WORD_SIZE, 5 * HEX_WORD_SIZE), 16),
-      validUntil: parseInt(element.slice(5 * HEX_WORD_SIZE, 6 * HEX_WORD_SIZE), 16),
-      isSellOrder: parseInt(element.slice(6 * HEX_WORD_SIZE, 7 * HEX_WORD_SIZE), 16) > 0,
-      priceNumerator: parseInt(element.slice(7 * HEX_WORD_SIZE, 8 * HEX_WORD_SIZE), 16),
-      priceDenominator: parseInt(element.slice(8 * HEX_WORD_SIZE, 9 * HEX_WORD_SIZE), 16),
-      remainingAmount: parseInt(element.slice(9 * HEX_WORD_SIZE, 10 * HEX_WORD_SIZE), 16),
+      user: "0x" + element.slice(0, 20 * 2), // address is only 20 bytes
```

Comment:
I'm not sure if all this arithmetic below here really helps comprehension (it confuses me). I'd just use the final numbers.

Reply:
I think it helps a lot as the same numbers also appear in the driver-code in rust.

Reply:
cute!

Reply:
Hmh, maybe a `pop` logic similar to what we are doing in the driver is easier to read?

Like:
```js
const user = parseInt(bytes.splice(0, ADDRESS_WIDTH))
const sellTokenBalance = parseInt(bytes.splice(0, UINT256_WIDTH))
...
```

Reply:
I like that I can see the indices like in the `from_bytes` in the driver code, but I do agree that it looks a bit repetitive to the point that I wouldn't read it anyway. Perhaps the most recent suggestion is best.